dali/pipeline/data/tensor_vector_test.cc

```diff
@@ -227,6 +232,43 @@ TYPED_TEST(TensorVectorSuite, MoveAssignmentMetaData) {
                                });
 }
 
+TYPED_TEST(TensorVectorSuite, DeviceIdPropagationMultiGPU) {
+  int num_devices = 0;
+  CUDA_CALL(cudaGetDeviceCount(&num_devices));
+  if (num_devices < 2) {
+    GTEST_SKIP() << "At least 2 devices needed for the test\n";
+  }
+  constexpr bool is_device = std::is_same_v<TypeParam, GPUBackend>;
+  constexpr bool is_pinned = !is_device;
+  AccessOrder order = AccessOrder::host();
+  TensorShape<> shape{42};
+  for (int device_id = 0; device_id < num_devices; device_id++) {
+    TensorVector<TypeParam> batch;
+    batch.SetSize(1);
+    DeviceGuard dg(device_id);
+    batch.set_device_id(device_id);
+    batch.set_pinned(is_pinned);
+    batch.set_type(DALI_UINT8);
+    batch.set_sample_dim(shape.sample_dim());
+    batch.set_order(order);
+    CUDA_CALL(cudaSetDevice(device_id));
```

Comment:
Isn't the DeviceGaurd calling the setdevice on its own?

Reply:
Fixing.